impl/src/main/java/org/jboss/weld/injection/InterceptionFactoryImpl.java

```diff
@@ -112,15 +118,41 @@ public T createInterceptedInstance(T instance) {
         }
         used = true;
 
-        if (annotatedType.getJavaClass().isInterface()) {
-            throw InterceptorLogger.LOG.interceptionFactoryNotOnInstance(annotatedType.getJavaClass().getCanonicalName());
+        AnnotatedType<T> originalAT = configurator != null ? configurator.complete() : annotatedType;
+        // this is the resulting AT we want to work with
+        AnnotatedType<T> resultingAnnotatedType = null;
+        // if it is an interface, we will have to 'merge' the AT with the AT of whatever user supplied as an instance
+        if (originalAT.getJavaClass().isInterface()) {
+            AnnotatedType<?> instanceAnnotatedType = beanManager.createAnnotatedType(instance.getClass());
+            AnnotatedTypeConfiguratorImpl<?> newAtConfigurator = new AnnotatedTypeConfiguratorImpl<>(instanceAnnotatedType);
+
+            // first check class-level annotation
+            for (Annotation annotation : originalAT.getAnnotations()){
+                if (isInterceptorBinding(annotation)){
+                    newAtConfigurator.add(annotation);
```

Comment:
I believe you should first check whether an annotation of the same annotated type is already present or not - interceptor bindings may have different values. In fact, different values should result in definition error.

Reply:
Yes, thought about it but for my first version, this was "good enough".
Do you happen to know if we have similar behaviour in, say, configurators?

Reply:
I don't think so - it's purely the responsibility of extensions ;-)

Reply:
Reminder to self - still have to add a test for this and then a fix.
Not entirely sure what is the 'correct' approach - we can either override anything user did (thinking that user actually wanted to do that) or go fool-proof and blow up if it goes against other annotations. Either way, we first need to detect it.